### PR TITLE
Gitlab token check

### PR DESCRIPTION
**Pull request by mappjzc**
### Summary
Add token permission check for connection test.

### Screenshots
image
image

### CODE REVIEW

File: backend/plugins/gitlab/api/connection.go [hunk 1]
```diff
@@ -49,11 +51,24 @@ func TestConnection(input *plugin.ApiResourceInput) (*plugin.ApiResourceOutput,
 		return nil, err
 	}
 
-	_, err = api.NewApiClientFromConnection(context.TODO(), basicRes, &connection)
+	apiClient, err := api.NewApiClientFromConnection(context.TODO(), basicRes, &connection)
 	if err != nil {
 		return nil, err
 	}
 
+	// check API/read_api permissions
```
Comment (abeizn) [inline backend/plugins/gitlab/api/connection.go:59]: https://devlake.apache.org/docs/next/Configuration/GitLab
image

Does this also verify read_user and read_repository  permission?

Comment (mappjzc) [reply backend/plugins/gitlab/api/connection.go:59]: I have tried it on my GitLab.
read_user + read_repository is not enough.
read_api can work success.

Comment (abeizn) [reply backend/plugins/gitlab/api/connection.go:59]: you are right, the docs need to update

**Review by abeizn — APPROVED** [review]
LGTM

**Pull request merged**